src/compiler/factory.ts

```diff
@@ -1618,17 +1618,23 @@ namespace ts {
         );
     }
 
-    function createReactNamespace(reactNamespace: string, parent: JsxOpeningLikeElement) {
+    export function createJsxFactory(jsxFactory: string) {
+        // No explicit validation of this parameter is required. Users are

```

Comment:
Maybe we should just restrict this to dotted names for simplicity's sake. Thoughts @mhegazy ?